compiler-core/src/build/package_compiler.rs

```diff
@@ -53,6 +54,7 @@ where
         lib: &'a Path,
         target: Target,
         ids: UniqueIdGenerator,
+        is_project_root: bool,
```

Comment:
The PackageCompiler doesn't have a concept of the root, it doesn't work at that level of abstraction, and in some uses of it (like the `compile-package` CLI) there is no such thing as a root.

Instead let's pass in a `Mode`, or set it as a property. This would be clearer than a boolean too.